game/core.py

```diff
@@ -251,11 +284,16 @@ def xmlrpc_query_state(self, request, game_id, player_token):
 
         # Game not begun, lobby state to be sent
         if curr_state is None:
+            if not game.num_AI:
+                game.add_AIplayer()
+                game.num_AI += 1
             result["game_state"] = "none"
             result["action"] = "wait"
             result["players"] = list(map(lambda x: x.alias, game.players))
 
         if curr_state is State.ROUND_CONT:
+            if(game.turn.auto):
+                temp = game.step(game.moveAI(game.turn))
```

Comment:
1. You can just return with None, State.ROUND_CONT after this. Or am I missing something? The problem right now is that you have effectively called evaluate inside evaluate, and I don't want that at all. There should be no recursive calls to evaluate ever. Figure out how to rig the state machine so that the next evaluate call happens in the next cycle.

2. Instead of temp, you can assign to "_", as in `_ = game.step(game.moveAI(game.turn))`, this is completely valid and safely ensures you're not using the LHS after this line

Reply:
I didn't get the first point. We are making changes in the query_state and not evaluate. And evaluate isn't getting called recursively?